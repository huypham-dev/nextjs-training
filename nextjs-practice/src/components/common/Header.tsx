import { memo } from "react";
import Image from "next/image";
import Link from "next/link";
import { NavigationItem } from "types";
import isEqual from "react-fast-compare";

type Props = {
  navigations: NavigationItem[];
};

const Header = ({ navigations }: Props) => (
  <header className="w-full mx-auto text-center">
    <Link href="/">
      <a className="inline-block text-2xl mb-12">
        <Image
          src="https://images.prismic.io/nextjsexample/ebd683dd-af14-49e3-bef5-5c497c9b8cac_logo.png"
          alt="logo"
          width={150}
          height={150}
        />
      </a>
    </Link>

    <ul className="flex justify-center b p-5 uppercase text-xs border-t border-b bg-gray-100">
      {navigations.map(({ label, path }) => (
        <li key={label}>
          <Link href={path}>
            <a className="text-gray-600 font-bold text-sm py-2 px-4 mx-2 hover:text-black">
              {label}
            </a>
          </Link>
        </li>
      ))}
    </ul>
  </header>
);

export default memo(Header, isEqual);
