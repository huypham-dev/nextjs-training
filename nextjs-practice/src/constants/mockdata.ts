export const CATEGORIES = [
  {
    id: "1",
    name: "mobile",
    image:
      "https://images.prismic.io/nextjsexample/7853f863-6a12-4312-a6da-f930437f6d62_mobile.png",
  },
  {
    id: "2",
    name: "internet",
    image:
      "https://images.prismic.io/nextjsexample/cb71ce70-bb03-442b-99e8-a318255b81f3_3579187.png",
  },
];

export const ARTICLES = [
  {
    id: "1",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
  {
    id: "2",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
];
