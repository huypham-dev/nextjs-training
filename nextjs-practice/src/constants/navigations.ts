export const navigations = [
  { label: "Home", path: "/" },
  { label: "Blog", path: "/blog" },
  { label: "Contact", path: "/contact" },
];
