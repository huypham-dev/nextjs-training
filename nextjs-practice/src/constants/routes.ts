export const ROUTES = {
  home: {
    path: "/",
  },
  blog: {
    path: "/blog",
    withId: (id: string) => `/blog/${id}`,
  },
  post: {
    path: "/post",
    withId: (id: string) => `/post/${id}`,
  },
  contact: {
    path: "/",
  },
};
