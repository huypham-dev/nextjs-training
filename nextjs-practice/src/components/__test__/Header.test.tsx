import { navigations } from "constants/mockdata";
import { mount, shallow } from "enzyme";
import Header from "../common/Header";

describe("Header", () => {
  it("Snapshot", () => {
    const component = shallow(<Header navigations={navigations} />);

    expect(component).toMatchSnapshot();
  });

  it("Render correctly with props", () => {
    const component = mount(<Header navigations={navigations} />);

    expect(component.prop("navigations")).toEqual(navigations);
    expect(component.find("li").length).toBe(3);
  });
});
