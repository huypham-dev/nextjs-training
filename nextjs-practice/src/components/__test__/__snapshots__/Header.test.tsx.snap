// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot 1`] = `
<header
  className="w-full mx-auto text-center"
>
  <Link
    href="/"
  >
    <a
      className="inline-block text-2xl mb-12"
    >
      <Image
        alt="logo"
        height={150}
        src="https://images.prismic.io/nextjsexample/ebd683dd-af14-49e3-bef5-5c497c9b8cac_logo.png"
        width={150}
      />
    </a>
  </Link>
  <ul
    className="flex justify-center b p-5 uppercase text-xs border-t border-b bg-gray-100"
  >
    <li
      key="Home"
    >
      <Link
        href="/"
      >
        <a
          className="text-gray-600 font-bold text-sm py-2 px-4 mx-2 hover:text-black"
        >
          Home
        </a>
      </Link>
    </li>
    <li
      key="Blog"
    >
      <Link
        href="/blog"
      >
        <a
          className="text-gray-600 font-bold text-sm py-2 px-4 mx-2 hover:text-black"
        >
          Blog
        </a>
      </Link>
    </li>
    <li
      key="Contact"
    >
      <Link
        href="/contact"
      >
        <a
          className="text-gray-600 font-bold text-sm py-2 px-4 mx-2 hover:text-black"
        >
          Contact
        </a>
      </Link>
    </li>
  </ul>
</header>
`;
